sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java

```diff
@@ -638,34 +652,132 @@ Mono<EncryptedBlob> encryptBlob(Flux<ByteBuffer> plainTextFlux) {
+
+                    return Flux.concat(Flux.just(ByteBuffer.wrap(gmcCipher.getIV())),
+                        cipherTextWithTag);
+                }, 1, 1);
+        return encryptedTextFlux;
+    }
+
+    private Flux<ByteBuffer> encryptV1(Flux<ByteBuffer> plainTextFlux, Cipher cbcCipher) {
+        Flux<ByteBuffer> encryptedTextFlux;
+        // Encrypt plain text with content encryption key
+        encryptedTextFlux = plainTextFlux.map(plainTextBuffer -> {
+            int outputSize = cbcCipher.getOutputSize(plainTextBuffer.remaining());
+
+            /*
+             * This should be the only place we allocate memory in encryptBlob(). Although there is an
+             * overload that can encrypt in place that would save allocations, we do not want to overwrite
+             * customer's memory, so we must allocate our own memory. If memory usage becomes unreasonable,
+             * we should implement pooling.
+             */
+            ByteBuffer encryptedTextBuffer = ByteBuffer.allocate(outputSize);
+
+            int encryptedBytes;
+            try {
+                encryptedBytes = cbcCipher.update(plainTextBuffer, encryptedTextBuffer);
+            } catch (ShortBufferException e) {
+                throw LOGGER.logExceptionAsError(Exceptions.propagate(e));
+            }
+            encryptedTextBuffer.position(0);
+            encryptedTextBuffer.limit(encryptedBytes);
+            return encryptedTextBuffer;
+        });
+
+        /*
+         * Defer() ensures the contained code is not executed until the Flux is subscribed to, in
+         * other words, cipher.doFinal() will not be called until the plainTextFlux has completed
+         * and therefore all other data has been encrypted.
+         */
+        encryptedTextFlux = Flux.concat(encryptedTextFlux,
+            Mono.fromCallable(() -> ByteBuffer.wrap(cbcCipher.doFinal())));
+        return encryptedTextFlux;
+    }
+
     SecretKey generateSecretKey() throws NoSuchAlgorithmException {
         KeyGenerator keyGen = KeyGenerator.getInstance(CryptographyConstants.AES);
         keyGen.init(256);
 
         return keyGen.generateKey();
     }
 
-    Cipher generateCipher(SecretKey aesKey) throws GeneralSecurityException {
-        Cipher cipher = Cipher.getInstance(CryptographyConstants.AES_CBC_PKCS5PADDING);
+    Cipher generateCBCCipher(SecretKey aesKey) throws GeneralSecurityException {
+        Cipher cipher = Cipher.getInstance(AES_CBC_PKCS5PADDING);
 
         // Generate content encryption key
         cipher.init(Cipher.ENCRYPT_MODE, aesKey);
 
         return cipher;
     }
 
+    Cipher generateGCMCipher(SecretKey aesKey, long index) throws NoSuchPaddingException, NoSuchAlgorithmException,
+        InvalidAlgorithmParameterException, InvalidKeyException {
+        Cipher cipher = Cipher.getInstance(AES_GCM_NO_PADDING);
+        byte[] iv = ByteBuffer.allocate(12).putLong(index).array(); // standard gcm nonce is 12 bytes per James' doc
```

Comment:
This shouldn't be a major concern, `ByteBuffer.putLong` is endian specific based on the `ByteBuffer` configuration or the default of the underlying system.

Reply:
Was there a change requested here? I agree that I don't think it should be a concern as long as we're symmetrical

Reply:
No change request, just a general callout of behavior in case we end up seeing something funny happen